### ISSUE TITLE
when set    bulkActionButtons={false} for  List ，the row height will become very dense

### ISSUE THREAD

**Issue by MoonSeeing** [marmelab/react-admin#3834]
**What you were expecting:**
 i want to hide bulkAction, no checkbox in  the  front of row.

**What happened instead:**
when i set bulkActionButtons={false}  in the List ,the table will became very dense.because the  checkbox is hide ,can not height up the row .i think the row height  should not depend whether have the checkbox.
i know i can overrides the material ui  in the theme to solve the problem
```
  overrides: {
    MuiTableRow: {
      root: {
   height: 40px;
        },
      },
    },
  },
```
but it not a good way ,so should solve it in the Datagrid.
**Steps to reproduce:**

1. download react-admin from github .
2. git checkout -b next .
3. make install  & make run-demo (now the demo is running by 3.0.0-alpha.0)
4. find  the file demo/src/reviews/ReviewList.js .
5. change the  code `bulkActionButtons={<ReviewsBulkActionButtons />}` to `  
  bulkActionButtons={false}`.
6.  Now  you can see the reviews list become dense.

**Related code:** 
 demo/src/reviews/ReviewList.js 
```
               <List
                           {...props}
                            className={classnames(classes.list, {
                              [classes.listWithDrawer]: isMatch,
                               })}
                               bulkActionButtons={false}
                            // bulkActionButtons={<ReviewsBulkActionButtons />}
                                filters={<ReviewFilter />}
                                perPage={25}
                                sort={{ field: 'date', order: 'DESC' }}
                            >
```

**Other information:**
before change the code
<img width="2328" alt="WeChat57b23ad7ba6e1dfcf533b5e998e811b5" src="https://user-images.githubusercontent.com/19420622/66996608-9d6ef000-f103-11e9-9c14-3ccfeb681a0d.png">

after  change the code
<img width="2210" alt="WeChatc4e82be2258499c5b563e2f3cd8d6311" src="https://user-images.githubusercontent.com/19420622/66996789-e9219980-f103-11e9-8ffc-6439600e0445.png">

**Environment**

* React-admin version:3.0.0-alpha.0

**Comment by fzaninotto** [comment]
Reproduced. The datagrid rows should be small when there is no checkbox, but not that small. 

Thanks for the bug report.

**Comment by fzaninotto** [comment]
Fixed by #3840
